### PR TITLE
Expose transferSize, encodedBodySize, decodedBodySize

### LINKED ISSUE #238: transferSize might reveal HttpOnly cookies
Which does not seem ideal. The web platform networking APIs purposefully filter out cookie headers everywhere. (Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1679737.)

### PR DESCRIPTION
**Pull request by noamr**
transferSize is based on the cache mode of the resource.
Closes #238

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/resource-timing/pull/266.html" title="Last updated on Apr 15, 2021, 6:02 AM UTC (27c8bda)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/resource-timing/266/7c535a9...27c8bda.html" title="Last updated on Apr 15, 2021, 6:02 AM UTC (27c8bda)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -485,6 +489,20 @@ <h3>The <dfn>PerformanceResourceTiming</dfn> Interface</h3>
...
+<p data-dfn-for="PerformanceResourceTiming">The <dfn>decodedBodySize</dfn> getter
+steps are to return <a>this</a>'s
+<a data-for="PerformanceResourceTiming">timing info</a>'s [=fetch timing info/decoded body size=].
+<p data-dfn-for="PerformanceResourceTiming">The <dfn>transferSize</dfn> getter
+steps are to perform the following steps:
+<ol>
+<li><p>If <a>this</a>'s <a data-for="PerformanceResourceTiming">cache mode</a> is
+"<code>local</code>", then return 0.
+<li><p>Return <a>this</a>'s
+<a data-for="PerformanceResourceTiming">timing info</a>'s [=fetch timing info/encoded body size=].
```
Comment (yoavweiss) [inline index.html:504]: If we add a fixed size here for the headers, we can close https://github.com/w3c/resource-timing/issues/238#issuecomment-762297269

File: index.html [hunk 2]
```diff
@@ -718,24 +736,26 @@ <h3>Resource Timing Attributes</h3>
...
 [=/fetch timing info=] |timingInfo|, a DOMString
-|requestedURL|, a DOMString |initiatorType| and a <a>global object</a> |global|:
+|requestedURL|, a DOMString |initiatorType| a <a>global object</a> |global|, and a string |cacheMode|,
+perform the following steps:
 <ol>
 <li>Create a <a>PerformanceResourceTiming</a> object |entry| in |global|'s [=global object/realm=].
-<li><a>Setup the resource timing entry</a> for |entry|, given |initiatorType|, |requestedURL| and
-|timingInfo|.
+<li><a>Setup the resource timing entry</a> for |entry|, given |initiatorType|, |requestedURL|,
+|timingInfo|, and the empty string or "<code>local</code>" |cacheMode|.
```
Comment (yoavweiss) [inline index.html:744]: If we're just passing |cacheMode|, I think you can drop the "empty string or local" part

File: index.html [hunk 3]
```diff
@@ -718,24 +736,26 @@ <h3>Resource Timing Attributes</h3>
 <h2>Creating a resource timing entry</h2>
 <p>To <dfn export="">mark resource timing</dfn> given a
 [=/fetch timing info=] |timingInfo|, a DOMString
-|requestedURL|, a DOMString |initiatorType| and a <a>global object</a> |global|:
+|requestedURL|, a DOMString |initiatorType| a <a>global object</a> |global|, and a string |cacheMode|,
```
Comment (yoavweiss) [inline index.html:739]: s/string/DOMString/ ?

File: index.html [hunk 4]
```diff
@@ -718,24 +736,26 @@ <h3>Resource Timing Attributes</h3>
...
 "PERFORMANCE-TIMELINE-2#dfn-queue-a-performanceentry">Queue</a> |entry|.</li>
 <li><a data-lt='add a PerformanceResourceTiming entry'>Add</a> |entry| to |global|'s
 <a data-cite="PERFORMANCE-TIMELINE-2#dfn-performance-entry-buffer">performance
 entry buffer</a>.</li>
 </ol>
 <p>To <dfn export="">setup the resource timing entry</dfn> for <a>PerformanceResourceTiming</a> |entry|
-given DOMStrring |initiatorType|, DOMString |requestedURL|, and [=/fetch timing info=] |timingInfo|,
-perform the following steps:</p>
+given DOMStrring |initiatorType|, DOMString |requestedURL|, [=/fetch timing info=] |timingInfo|, and
+the empty string or "<code>local</code>" |cacheMode|, perform the following steps:</p>
```
Comment (yoavweiss) [inline index.html:753]: I think you can replace the possible values with "DOMString", and maybe assert that it's one of the possible values below.

File: index.html (hunk 3 above)
Comment (yoavweiss) [reply index.html:739]: Also, are you planning to change the callers?

**Review by yoavweiss — COMMENTED** [review]
Thanks for pushing this!

File: index.html (hunk 1 above)
Comment (noamr) [reply index.html:504]: Sure, I can do that, what value should I add?

File: index.html (hunk 3 above)
Comment (noamr) [reply index.html:739]: Already changed FETCH (Waiting for review): https://github.com/whatwg/fetch/pull/1202. Will modify  HTML for nav-timing afterwards.

File: index.html [hunk 5]
```diff
@@ -718,24 +737,27 @@ <h3>Resource Timing Attributes</h3>
...
 [=/fetch timing info=] |timingInfo|, a DOMString
-|requestedURL|, a DOMString |initiatorType| and a <a>global object</a> |global|:
+|requestedURL|, a DOMString |initiatorType| a <a>global object</a> |global|, and a string |cacheMode|,
+perform the following steps:
 <ol>
 <li>Create a <a>PerformanceResourceTiming</a> object |entry| in |global|'s [=global object/realm=].
-<li><a>Setup the resource timing entry</a> for |entry|, given |initiatorType|, |requestedURL| and
-|timingInfo|.
+<li><a>Setup the resource timing entry</a> for |entry|, given |initiatorType|, |requestedURL|,
+|timingInfo|, |cacheMode|.
```
Comment (yoavweiss) [inline index.html:745]: s/,/, and/

File: index.html [hunk 6]
```diff
@@ -485,6 +489,21 @@ <h3>The <dfn>PerformanceResourceTiming</dfn> Interface</h3>
...
+steps are to return <a>this</a>'s
+<a data-for="PerformanceResourceTiming">timing info</a>'s [=fetch timing info/decoded body size=].
+<p data-dfn-for="PerformanceResourceTiming">The <dfn>transferSize</dfn> getter
+steps are to perform the following steps:
+<ol>
+<li><p>If <a>this</a>'s <a data-for="PerformanceResourceTiming">cache mode</a> is
+"<code>local</code>", then return 0.
+<li><p>Return <a>this</a>'s
+<a data-for="PerformanceResourceTiming">timing info</a>'s [=fetch timing info/encoded body size=]
+plus 300.
```
Comment (yoavweiss) [inline index.html:505]: Can you add a note indicating why we're adding "300" here?

**Review by yoavweiss — APPROVED** [review]
LGTM % comments

**Comment by noamr** [comment]
PR requests addressed. Can merge?

**Review by yoavweiss — APPROVED** [review]
LGTM

**Review by liuchengwei555 — COMMENTED** [review]
C

**Pull request merged**